### ISSUE TITLE
JSX Not validating when index signature is present

### ISSUE THREAD

**Issue by bezreyhan** [Microsoft/TypeScript#10265]
**TypeScript Version:**  2.0

**Code**

```ts
interface IProps {
  primaryText: string,
  [propName: string]: any
}

function VerticalNavMenuItem(props: IProps) {
  return <div>props.primaryText</div>
}

function VerticalNav() {
  return (
    <div>
      <SOME_JSX />
      <VerticalNavMenuItem primaryText={2} />
    </div>
  )
} 
```

**Expected behavior:**
In VerticalNav I have the child component: `<VerticalNavMenuItem primaryText={2} />`
This should throw a type error because primary text must be a string.

However, if I remove `  [propName: string]: any` from IProps then the type error will be thrown.

**Actual behavior:** 
No error is thrown.

**Comment by RyanCavanaugh** [comment]
The interpreted intent of the index signature is to allow arbitrary extra properties.

**Comment by bezreyhan** [comment]
@RyanCavanaugh thanks for the quick response. I guess I'm confused - shouldn't the index signature allow any additional properties with `any` type, but nevertheless check that the primaryText property has the type `string`?

Maybe I am misunderstanding the docs but for the following example: 
```
interface SquareConfig {
    color?: string;
    width?: number;
    [propName: string]: any;
}
```
the docs say that: `SquareConfig can have any number of properties, and as long as they aren’t color or width, their types don’t matter.`

Does that not mean that the types for `color` and `width` do matter?

Also, when I tested this in the TS playground using normal functions the primaryText was checked to be a string.

**Comment by RyanCavanaugh** [comment]
Sorry, I totally misread your original post. Rereading/investigating

**Comment by RyanCavanaugh** [comment]
Yeah this is totally supposed to be an error. Thanks for following up!
